[CSS Zoom] Apply zoom factor to text-decoration-thickness

brentfulgham opened the pull request:
#### a4b32d383c7bc91cd7b8e124788afc0184eae260
<pre>
[CSS Zoom] Apply zoom factor to text-decoration-thickness
<a href="https://bugs.webkit.org/show_bug.cgi?id=300642">https://bugs.webkit.org/show_bug.cgi?id=300642</a>
&lt;<a href="https://rdar.apple.com/problem/162543798">rdar://problem/162543798</a>&gt;

Reviewed by Tim Nguyen.

Adapt the text-decoration-thickness logic to the new CSS Zoom architecture.

Tests: imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style.html
       imported/w3c/web-platform-tests/css/css-viewport/zoom/text-decoration-thickness.html

* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/reference/text-decoration-thickness-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/text-decoration-thickness-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/text-decoration-thickness.html: Added.
* Source/WebCore/rendering/TextBoxPainter.cpp:
(WebCore::computedTextDecorationThickness):
(WebCore::computedAutoTextDecorationThickness):
* Source/WebCore/style/InlineTextBoxStyle.cpp:
(WebCore::computedInkOverflowForDecorations):
* Source/WebCore/style/values/text-decoration/StyleTextDecorationThickness.cpp:
(WebCore::Style::TextDecorationThickness::resolve const):
* Source/WebCore/style/values/text-decoration/StyleTextDecorationThickness.h:

Canonical link: <a href="https://commits.webkit.org/301664@main">https://commits.webkit.org/301664@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9068cabb9bab4cda1a54a1f270e43c34b9ad9818

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126671 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46314 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37277 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133634 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/78323 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128542 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46946 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54851 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/96395 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/64456 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/fc57cb96-b097-4a0b-bd09-258122c19491) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129618 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37561 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113293 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/76920 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/92c70478-cfba-4cf0-9f65-344003fd41ca) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/36447 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/31478 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77030 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/107376 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/31775 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136208 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53356 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/41053 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/104910 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53846 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109652 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/104611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26676 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50100 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28435 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50791 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53290 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59086 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/52550 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55886 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54290 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
Code change looks correct. I would just add tests:

* Rendering: `css/css-viewport/zoom/text-underline-offset.html` is a good one to go off
* SVG computed style: you can just add an entry to `css/css-viewport/zoom/svg-computed-style.html` (similar to underline-offset)

nt1m reviewed (commented):
all the test coverage got lost :( and it's reflected in the commit message with "No new tests"

brentfulgham committed:
[CSS Zoom] Apply zoom factor to text-decoration-thickness
https://bugs.webkit.org/show_bug.cgi?id=300642
<rdar://problem/162543798>

Reviewed by Tim Nguyen.

Adapt the text-decoration-thickness logic to the new CSS Zoom architecture.

Tests: imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style.html
       imported/w3c/web-platform-tests/css/css-viewport/zoom/text-decoration-thickness.html

* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/reference/text-decoration-thickness-ref.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/svg-computed-style.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/text-decoration-thickness-expected.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/css-viewport/zoom/text-decoration-thickness.html: Added.
* Source/WebCore/rendering/TextBoxPainter.cpp:
(WebCore::computedTextDecorationThickness):
(WebCore::computedAutoTextDecorationThickness):
* Source/WebCore/style/InlineTextBoxStyle.cpp:
(WebCore::computedInkOverflowForDecorations):
* Source/WebCore/style/values/text-decoration/StyleTextDecorationThickness.cpp:
(WebCore::Style::TextDecorationThickness::resolve const):
* Source/WebCore/style/values/text-decoration/StyleTextDecorationThickness.h:

Canonical link: https://commits.webkit.org/301664@main
